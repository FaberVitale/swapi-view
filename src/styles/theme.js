//@flow
import {
  yellow200 as primary1,
  yellow500 as primary2,
  grey400 as accent3
} from "material-ui/styles/colors";
import darkBaseTheme from "material-ui/styles/baseThemes/darkBaseTheme";
import getMuiTheme from "material-ui/styles/getMuiTheme";

const boxShadow =
  "0 1px 6px rgba(0, 0, 0, 0.12), 0 1px 4px rgba(0, 0, 0, 0.12)";

const theme = getMuiTheme(darkBaseTheme, {
  palette: {
    primary1Color: primary1,
    primary2Color: primary2,
    accent3Color: accent3,
    accent1Color: primary1,
    alternateTextColor: darkBaseTheme.palette.canvasColor
  },
  topBar: {
    backgroundColor: darkBaseTheme.palette.canvasColor,
    padding: 0,
    margin: 0,
    boxShadow: boxShadow
  },

  paper: {
    boxShadow
  },

  homePaper: {
    backgroundColor: "#000000"
  },

  extLink: {
    color: primary1,
    outlineColor: primary2
  },

  h1: {
    color: primary2,
    fontSize: 24
  },
  h2: {
    color: primary2
  },

  noResults: {
    color: "rgba(255, 255, 255, .5)"
  },

  autocomplete: {
    icon: {
      fill: primary1
    }
  },

  textField: {
    floatingLabelColor: "#fefefe",
    focusColor: primary2,
    borderColor: "#fefefe",
    hintColor: "rgba(255, 255, 255, .5)"
  },

  dropDownMenu: {
    accentColor: "#fefefe",
    focusColor: primary2
  },

  h3: darkBaseTheme.h4,

  p: {
    color: primary2
  },
  failedToLoad: {
    frown: {
      color: primary2,
      opacity: 0.1
    }
  },
  menuItem: {
    selectedTextColor: primary2
  }
});

export default theme;
