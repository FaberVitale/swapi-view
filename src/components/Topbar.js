//@flow
import React, { PureComponent } from "react";
import { connectDrawer as connect } from "../util/reactReduxConfig";
import Menu from "material-ui/svg-icons/navigation/menu";
import IconButton from "material-ui/IconButton";
import theme from "../styles/theme";

type Props = {
  toggleDrawer: (val: boolean) => void,
  isOpen: boolean
};
/* An AppBar that has a left menu icon and a title next to it,
 * its named export is just for testing and
 * its button opens the drawer
 */
export class Topbar extends PureComponent<Props> {
  onIconClick = (e: SyntheticMouseEvent<*>) => {
    e.preventDefault();
    this.props.toggleDrawer(true);
  };

  render() {
    return (
      <header
        className="topbar justify-space-between align-items-center"
        style={theme.topBar}
        aria-hidden={this.props.isOpen}
      >
        <div className="flex-row align-items-center justify-flex-start">
          <IconButton
            aria-label="open menu"
            aria-expanded={this.props.isOpen}
            onClick={this.onIconClick}
          >
            <Menu />
          </IconButton>
          <h1 style={theme.h1}>Swapi View</h1>
        </div>
      </header>
    );
  }
}

export default connect(true)(Topbar);
