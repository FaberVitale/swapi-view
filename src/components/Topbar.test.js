import React from "react";
import { Topbar } from "./Topbar";
import Renderer from "react-test-renderer";
import theme from "../styles/theme";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

describe("components/Topbar", () => {
  const toggleDrawer = jest.fn();

  const close = Renderer.create(
    <MuiThemeProvider muiTheme={theme}>
      <Topbar isOpen={false} toggleDrawer={toggleDrawer} />
    </MuiThemeProvider>
  ).toJSON();

  const open = Renderer.create(
    <MuiThemeProvider muiTheme={theme}>
      <Topbar isOpen={true} />
    </MuiThemeProvider>
  ).toJSON();

  it("matches the snapahot", () => {
    expect(close).toMatchSnapshot();

    expect(open).toMatchSnapshot();
  });
});
