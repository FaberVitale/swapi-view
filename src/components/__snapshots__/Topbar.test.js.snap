// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Topbar matches the snapahot 1`] = `
<header
  aria-hidden={false}
  className="topbar justify-space-between align-items-center"
  style={
    Object {
      "backgroundColor": "#303030",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12), 0 1px 4px rgba(0, 0, 0, 0.12)",
      "margin": 0,
      "padding": 0,
    }
  }
>
  <div
    className="flex-row align-items-center justify-flex-start"
  >
    <button
      aria-expanded={false}
      aria-label="open menu"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
    </button>
    <h1
      style={
        Object {
          "color": "#ffeb3b",
          "fontSize": 24,
        }
      }
    >
      Swapi View
    </h1>
  </div>
</header>
`;

exports[`components/Topbar matches the snapahot 2`] = `
<header
  aria-hidden={true}
  className="topbar justify-space-between align-items-center"
  style={
    Object {
      "backgroundColor": "#303030",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12), 0 1px 4px rgba(0, 0, 0, 0.12)",
      "margin": 0,
      "padding": 0,
    }
  }
>
  <div
    className="flex-row align-items-center justify-flex-start"
  >
    <button
      aria-expanded={true}
      aria-label="open menu"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
    </button>
    <h1
      style={
        Object {
          "color": "#ffeb3b",
          "fontSize": 24,
        }
      }
    >
      Swapi View
    </h1>
  </div>
</header>
`;
